### ISSUE TITLE
Nightly smoke test discrepancy with use case scenario

### ISSUE THREAD

**Issue by mthrok** [pytorch/audio#858]
Past couple of days, the nightly smoke tests of linux conda are failing with "segmentation fault", however when I perform the same operation (`python -c "import torchaudio"`) with a distributed nightly built in a fresh environment, it works.

<Details><Summary>install</Summary>

    ```
    $ conda create -q -y --name foobar python=3.6 && conda activate foobar
    Collecting package metadata (current_repodata.json): ...working... done
    Solving environment: ...working... done
    
    ## Package Plan ##
    
      environment location: /home/moto/conda/envs/foobar
    
      added / updated specs:
        - python=3.6
    
    
    The following NEW packages will be INSTALLED:
    
      _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main
      ca-certificates    pkgs/main/linux-64::ca-certificates-2020.6.24-0
      certifi            pkgs/main/linux-64::certifi-2020.6.20-py36_0
      ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.33.1-h53a641e_7
      libedit            pkgs/main/linux-64::libedit-3.1.20191231-h14c3975_1
      libffi             pkgs/main/linux-64::libffi-3.3-he6710b0_2
      libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.1.0-hdf63c60_0
      libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.1.0-hdf63c60_0
      ncurses            pkgs/main/linux-64::ncurses-6.2-he6710b0_1
      openssl            pkgs/main/linux-64::openssl-1.1.1g-h7b6447c_0
      pip                pkgs/main/linux-64::pip-20.1.1-py36_1
      python             pkgs/main/linux-64::python-3.6.10-h7579374_2
      readline           pkgs/main/linux-64::readline-8.0-h7b6447c_0
      setuptools         pkgs/main/linux-64::setuptools-49.2.0-py36_0
      sqlite             pkgs/main/linux-64::sqlite-3.32.3-h62c20be_0
      tk                 pkgs/main/linux-64::tk-8.6.10-hbc83047_0
      wheel              pkgs/main/linux-64::wheel-0.34.2-py36_0
      xz                 pkgs/main/linux-64::xz-5.2.5-h7b6447c_0
      zlib               pkgs/main/linux-64::zlib-1.2.11-h7b6447c_3
    
    
    Preparing transaction: ...working... done
    Verifying transaction: ...working... done
    Executing transaction: ...working... done
    
    $ conda install -q -y torchaudio -c pytorch-nightly
    Collecting package metadata (current_repodata.json): ...working... done
    Solving environment: ...working... done
    
    ## Package Plan ##
    
      environment location: /home/moto/conda/envs/foobar
    
      added / updated specs:
        - torchaudio
    
    
    The following NEW packages will be INSTALLED:
    
      blas               pkgs/main/linux-64::blas-1.0-mkl
      cudatoolkit        pkgs/main/linux-64::cudatoolkit-10.2.89-hfd86e86_1
      intel-openmp       pkgs/main/linux-64::intel-openmp-2020.1-217
      mkl                pkgs/main/linux-64::mkl-2020.1-217
      mkl-service        pkgs/main/linux-64::mkl-service-2.3.0-py36he904b0f_0
      mkl_fft            pkgs/main/linux-64::mkl_fft-1.1.0-py36h23d657b_0
      mkl_random         pkgs/main/linux-64::mkl_random-1.1.1-py36h0573a6f_0
      ninja              pkgs/main/linux-64::ninja-1.10.0-py36hfd86e86_0
      numpy              pkgs/main/linux-64::numpy-1.19.1-py36hbc911f0_0
      numpy-base         pkgs/main/linux-64::numpy-base-1.19.1-py36hfa32c7d_0
      pytorch            pytorch-nightly/linux-64::pytorch-1.7.0.dev20200805-py3.6_cuda10.2.89_cudnn7.6.5_0
      six                pkgs/main/noarch::six-1.15.0-py_0
      torchaudio         pytorch-nightly/linux-64::torchaudio-0.7.0.dev20200805-py36
      typing_extensions  pkgs/main/noarch::typing_extensions-3.7.4.2-py_0
    
    
    Preparing transaction: ...working... done
    Verifying transaction: ...working... done
    Executing transaction: ...working... done
    ```

</Details>
<Details><Summary>env info</Summary>

    ```
    $ python -m torch.utils.collect_env
    Collecting environment information...
    PyTorch version: 1.7.0.dev20200805
    Is debug build: No
    CUDA used to build PyTorch: 10.2
    
    OS: Ubuntu 18.04.3 LTS
    GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
    CMake version: version 3.10.2
    
    Python version: 3.6
    Is CUDA available: No
    CUDA runtime version: 10.1.243
    GPU models and configuration:
    GPU 0: Quadro GP100
    GPU 1: Quadro GP100
    
    Nvidia driver version: 418.116.00
    cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
    
    Versions of relevant libraries:
    [pip3] numpy==1.19.1
    [pip3] torch==1.7.0.dev20200805
    [pip3] torchaudio==0.7.0a0+4463fbd
    [conda] blas                      1.0                         mkl
    [conda] cudatoolkit               10.2.89              hfd86e86_1
    [conda] mkl                       2020.1                      217
    [conda] mkl-service               2.3.0            py36he904b0f_0
    [conda] mkl_fft                   1.1.0            py36h23d657b_0
    [conda] mkl_random                1.1.1            py36h0573a6f_0
    [conda] numpy                     1.19.1           py36hbc911f0_0
    [conda] numpy-base                1.19.1           py36hfa32c7d_0
    [conda] pytorch                   1.7.0.dev20200805 py3.6_cuda10.2.89_cudnn7.6.5_0    pytorch-nightly
    [conda] torchaudio                0.7.0.dev20200805            py36    pytorch-nightly
    ```

</Details>
<Details><Summary>smoke test</Summary>

    ```
    $ python -c 'import torchaudio;print(torchaudio.__version__)'
    0.7.0a0+4463fbd
    ```

</Details>

This suggests that the nightly smoke tests `torchaudio` CI is running do not reflect the real world use case. My brief understanding is that in CI, artifact objects are passed down between stages. (and also there is an issue with the way package is installed, which was only resolved in https://github.com/pytorch/text/pull/803).

@seemethere Once the binary built is uploaded, I think installing the distributed binary and using it for smoke test suits better for the purpose of package validation. What do you think? Is there an advantage of the current method?

**Comment by seemethere** [comment]
For smoke tests it should be unnecessary to actually upload the built binary. I think in reality the ideal way to do these tests is to smoke test first and upload only after the smoke test is actually done.

Did some digging, the actual cause of the segmentation fault is that the smoke test is not actually installing the nightly version but installing the latest version on `pytorch-test`, I'll have a follow up PR to clean this up
